src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java

```diff
@@ -1124,6 +1297,192 @@ protected Name readSystemPropertyName() {
+                        if (count(ch) == count) {
+                            return bp;
+                        }
+                    }
+                    nextChar();
+                }
+
+            }
+        }
+        // found end of input
+        return -1;
+    }
+
+    /**
+     * The kinds of line, used when reading a Markdown documentation comment.
+     * The values are loosely ordered by a combination of the specificity of
+     * the associated pattern, and the likelihood of use in a doc comment.
+     *
+     * Note that some of the patterns overlap, such as SETEXT_UNDERLINE and
+     * THEMATIC_BREAK. The CommonMark spec resolves the ambiguities with
+     * priority rules, such as the following:
+     *
+     * * If a line of dashes that meets the above conditions for being a
+     *   thematic break could also be interpreted as the underline of a
+     *   setext heading, the interpretation as a setext heading takes precedence.
+     *
+     * * When both a thematic break and a list item are possible interpretations
+     *   of a line, the thematic break takes precedence.
+     *
+     * Here in this context, the primary purpose of these kinds is to help
+     * determine when a block boundary terminates literal text, such as a
+     * code span, and so the exact nature of the line kind does not matter.
+     */
+    enum LineKind {
+        BLANK(Pattern.compile("[ \t]*")),
+
+        /**
+         * ATX header: starts with 1 to 6 # characters, followed by space or tab or end of line.
+         * @see <a href="https://spec.commonmark.org/0.30/#atx-headings">ATX Headings</a>
+         */
+        ATX_HEADER(Pattern.compile("#{1,6}([ \t].*|$)")),
+
+        /**
+         * Setext header: underline is sequence of = or - followed by optional spaces and tabs.
+         * @see <a href="https://spec.commonmark.org/0.30/#setext-headings">Setext Headings</a>
+         */
+        SETEXT_UNDERLINE(Pattern.compile("(=+|-+)[ \t]*")),
+
+        /**
+         * Thematic break: a line of * - _ interspersed with optional spaces and tabs
+         * @see <a href="https://spec.commonmark.org/0.30/#thematic-breaks">Thematic Break</a>
+         */
+        THEMATIC_BREAK(Pattern.compile("((\\*[ \t]*+){3,})|((-[ \t]*+){3,})|((_[ \t]*+){3,})")),
+
+        /**
+         * Code fence: 3 or more back ticks or tildes; back tick fence cannot have back ticks
+         * in the info string.
+         * Note potential conflict with strikeout for similar reasons if strikeout is supported.
+         * @see <a href="https://spec.commonmark.org/0.30/#code-fence">Code Fence</a>
+         */
+        CODE_FENCE(Pattern.compile("(`{3,}[^`]*+)|(~{3,}.*+)")),
+
+        /**
+         * Bullet list item: * + - followed by a space
+         * @see <a href="https://spec.commonmark.org/0.30/#list-items">List items</a>
+         */
+        BULLETED_LIST_ITEM(Pattern.compile("[-+*] .*")),
```

Comment:
This comment is about `BULLETED_LIST_ITEM` and `ORDERED_LIST_ITEM` constants. I know that we don't need to be very precise, but perhaps in this case we should. While the CommonMark spec is a vague on that, from my experiments with [dingus](https://spec.commonmark.org/dingus/), it appears that a list marker can be preceded and followed by some number of whitespace characters.

Reply:
whitespace is handled separately, on line 280 (`readIndent`) and285 (`: (indent <= 3) ? peekLineKind()`)

Reply:
Correct, but I believe the ordered list marker should be like this:

    ORDERED_LIST_ITEM(Pattern.compile("[0-9]{1,9}[.)] .*"))
                                                     ^

Note extra whitespace character. I think we should really add this to our tests, since lists are foundational Markdown structures, probably right after italics and bold.

Reply:
My recollection is that the space is required.   I will double check.

Reply:
Although it does not seem to be covered in the obvious place in the spec.   But see here:

```
Note that at least one space or tab is needed between the list marker and any following content, so these are not list items:

[Example 261](https://spec.commonmark.org/0.30/#example-261)Try It
```

I found the need for a space when playing with the dingus.   That text (which I just found) suggests that we need to allow for tab as well.

Reply:
Then we should add `[ \t]` to both constants, right:

    BULLETED_LIST_ITEM(Pattern.compile("[-+*][ \t].*"))
    ORDERED_LIST_ITEM(Pattern.compile("[0-9]{1,9}[.)][ \t].*"))